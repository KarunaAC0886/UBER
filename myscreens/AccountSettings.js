import { View, Text, StatusBar, Image, TouchableOpacity } from 'react-native'
import React from 'react'
import BackNavigator from './BackNavigator'
import styles from './styles'
import {FontAwesome5} from '@expo/vector-icons'
const AccountSettings = () => {
    return (
        <View style={styles.AccSetngs_main_cont}>
            <BackNavigator titleName='Account Settings' navpath="account"/>
            <View style={styles.AccSetngs_cont}>
                <TouchableOpacity style={styles.AccSetngs_prfl_cont}>
                    <Image style={styles.AccSetngs_prfl_img} source={require('../assets/profile.png')} />
                    <View style={styles.AccSetngs_prfl_contnt}>
                        <Text style={styles.AccSetngs_prfl_contnt_nm}>Krishna</Text>
                        <View style={styles.AccSetngs_prfl_contnt_pncntr}>
                            <Text style={styles.AccSetngs_prfl_contnt_pnfrnt}>+91</Text><Text style={styles.AccSetngs_prfl_contnt_pnbck}>81234 56789</Text>
                        </View>
                    </View>
                </TouchableOpacity>
                <View style={styles.AccSetngs_favrts_cont}>
                    <Text style={styles.AccSetngs_favrts_lbl}>Favorites</Text>
                    <View style={styles.AccSetngs_work_contr}>
                        <TouchableOpacity style={styles.AccSetngs_work_ttl}>
                            <FontAwesome5 name='home' style={styles.AccSetngs_work_homeicn} />
                            <Text style={styles.AccSetngs_work_homelbl}>Add Home</Text>
                        </TouchableOpacity>
                    </View>
                    <View style={styles.AccSetngs_work_contr}>
                        <TouchableOpacity style={styles.AccSetngs_work_ttl}>
                            <FontAwesome5 name='suitcase' style={styles.AccSetngs_work_homeicn} />
                            <Text style={styles.AccSetngs_work_homelbl}>Add Work</Text>
                        </TouchableOpacity>
                    </View>
    
                </View>
            </View>
            <StatusBar />
        </View>
    )
}

export default AccountSettings