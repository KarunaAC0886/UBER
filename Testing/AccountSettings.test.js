import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { useNavigation } from '@react-navigation/native';
import AccountSettings from '../myscreens/AccountSettings';

jest.mock('@expo/vector-icons', () => ({
    FontAwesome5: 'FontAwesome5',
}));


jest.mock('@react-navigation/native', () => ({
  useNavigation: jest.fn(),
}));
jest.mock('react-native-responsive-screen', () => ({
  widthPercentageToDP: jest.fn(),
  heightPercentageToDP: jest.fn(),
}));
describe('AccountSettings', () => {
  it('AccountSettings page renders correctly', () => {
   
    useNavigation.mockReturnValue({ navigate: jest.fn() });


    const tree = renderer.create(<AccountSettings/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});