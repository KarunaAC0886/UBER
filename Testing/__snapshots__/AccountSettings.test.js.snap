// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountSettings AccountSettings page renders correctly 1`] = `
<View
  style={
    {
      "height": "100%",
      "width": "100%",
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "#000000",
        "height": "13%",
        "justifyContent": "space-between",
        "paddingBottom": "5%",
        "paddingHorizontal": "6%",
        "paddingVertical": "2%",
        "width": "100%",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <FontAwesome5
        name="arrow-left"
        style={
          {
            "color": "#ffffff",
            "fontSize": 20,
          }
        }
      />
    </View>
    <Text
      style={
        {
          "color": "#ffffff",
          "fontFamily": "Roboto",
          "fontSize": 36,
        }
      }
    >
      Account Settings
    </Text>
  </View>
  <View
    style={
      {
        "backgroundColor": "#1A1A1A",
        "height": "90%",
        "width": "100%",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "flex": 0,
          "flexDirection": "row",
          "height": "15%",
          "opacity": 1,
          "paddingHorizontal": "6%",
          "paddingTop": "6%",
          "width": "100%",
        }
      }
    >
      <Image
        source={
          {
            "testUri": "../../../assets/profile.png",
          }
        }
        style={
          {
            "height": 80,
            "width": 80,
          }
        }
      />
      <View
        style={
          {
            "paddingLeft": "5%",
          }
        }
      >
        <Text
          style={
            {
              "color": "#ffffff",
              "fontSize": 22,
            }
          }
        >
          Krishna
        </Text>
        <View
          style={
            {
              "flex": 0,
              "flexDirection": "row",
              "paddingTop": "2%",
            }
          }
        >
          <Text
            style={
              {
                "color": "#EDF6FF",
                "fontSize": 19,
              }
            }
          >
            +91
          </Text>
          <Text
            style={
              {
                "color": "#EDF6FF",
                "fontSize": 19,
                "paddingLeft": "2%",
              }
            }
          >
            81234 56789
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "flex": 0,
          "flexDirection": "column",
          "height": "auto",
          "paddingLeft": "7%",
          "paddingTop": 10,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontSize": 18,
            "fontWeight": "400",
          }
        }
      >
        Favorites
      </Text>
      <View
        style={
          {
            "height": "auto",
            "paddingTop": 25,
            "width": "100%",
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "flex": 0,
              "flexDirection": "row",
              "opacity": 1,
            }
          }
        >
          <FontAwesome5
            name="home"
            style={
              {
                "color": "#ffffff",
                "fontSize": 18,
              }
            }
          />
          <Text
            style={
              {
                "color": "#DADADA",
                "fontSize": 18,
                "paddingLeft": "4%",
              }
            }
          >
            Add Home
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "height": "auto",
            "paddingTop": 25,
            "width": "100%",
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "flex": 0,
              "flexDirection": "row",
              "opacity": 1,
            }
          }
        >
          <FontAwesome5
            name="suitcase"
            style={
              {
                "color": "#ffffff",
                "fontSize": 18,
              }
            }
          />
          <Text
            style={
              {
                "color": "#DADADA",
                "fontSize": 18,
                "paddingLeft": "4%",
              }
            }
          >
            Add Work
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
